src/app/data/services/local_storage_service.ts

```diff
@@ -0,0 +1,30 @@
+///<reference path="../models/content_block.ts"/>
+///<reference path="ajax_content_blocks_hash_serializer.ts"/>
+
+class LocalStorageService {
+  static EXPIRATIONTIME = 2592000000; //ms (30 days)
```

Comment:
5 minutes would actually be OK. if the data has already been loaded by ajax, we can also store it here. i would set it to somewhere between 30 minutes and 1 day, but 30 days is imo a bit too much